### ISSUE TITLE
Error: The element has no supported sources

### ISSUE THREAD

**Issue by daniel347x** [chrisurf/obsidian-voice#2]
I created a new user/role in my AWS account with full "Polly" access, but nothing else. Then, I created the access key/secret for this user and added to Voice settings in Obsidian.

When I attempted to run 'Voice: Start reading the current document" I receive the following error (visible in the console):

"plugin:voice:7240 Uncaught (in promise) DOMException: The element has no supported sources."

Any suggestions?

**Comment by chrisurf** [comment]
@daniel347x , have you tried to entitle your IAM user with the AWS managed policy `AmazonPollyFullAccess`?
If the answer is yes: Have you tried the following steps in the following order?

1. Have added the correct aws access key and aws access key secret?
2. Have you selected an aws region where that IAM user is entitled to?
3. Have you closed Obsidian, restarted it Obsidian and then verified that the correct credentials are still configured?
4. Have you tested it this with a very small test text example (to ensure it is not a connectivity/latency issue)?

If you still seeing the same error: Provide more context, such as device, os, os version.

**Comment by splunker691** [comment]
Any chance you will make it possible to use other voice sources? 
Besides AWS being painful to navigate (I've had an AWS account for years but always end up giving up after it sends me in circles). But also, for some writers they use voice as part of the proofreading stage (I do) and doing this will chew through the free characters on AWS pretty quick. 
Currently Office 365 has a free voice function within docs that works perfectly for the review stage. Though I can't speak for other writers, I can tell you that I don't need a fancy voice for this process.
What would AWS would be amazing for would be for outputting an audio book. I would totally make the effort to figure out how to get what I needed from AWS for the ability to do output an audio book.
Juist a thought (or two).

**Comment by chrisurf** [comment]
> splunker691 @splunker691 It appears that your comment is not directly relate to the current issue.

Nevertheless, at this point, I'm prioritizing my focus on AWS API.

**Comment by sebastianhuus** [comment]
Hi! Seeing the same error. 

Obsidian 1.4.16 on MacOS Sonoma 14.1.
At the time of writing, I see that the voice I selected in Obsidian is not available for my Region (eu-north). I fixed this but the DOM error still remains.

Log:
```
plugin:voice:7241 Uncaught (in promise) DOMException: The element has no supported sources.
polly.eu-north-1.amazonaws.com/v1/speech:1     Failed to load resource: the server responded with a status of 400 (Bad Request)
plugin:voice:7228 Error playing the audio stream: ValidationException: The selected engine is not supported in this region
    at eval (plugin:voice:3980:51)
    at step (plugin:voice:2925:19)
    at Object.eval [as next] (plugin:voice:2872:14)
    at fulfilled (plugin:voice:2843:24)
callPolly @ plugin:voice:7228
2plugin:voice:7241 Uncaught (in promise) DOMException: The element has no supported sources.
```

Log after fixing the region issue:
```
plugin:voice:7241 Uncaught (in promise) DOMException: The element has no supported sources.
  playAudio @ plugin:voice:7241
  callback @ plugin:voice:7527
  rj @ app.js:1
  t.onChooseItem @ app.js:1
  t.onChooseSuggestion @ app.js:1
  t.selectSuggestion @ app.js:1
  e.useSelectedItem @ app.js:1
  (anonymous) @ app.js:1
  e.handleKey @ app.js:1
  e.onKeyEvent @ app.js:1
```

**Comment by chrisurf** [comment]
> Hi! Seeing the same error.
> 
> Obsidian 1.4.16 on MacOS Sonoma 14.1. At the time of writing, I see that the voice I selected in Obsidian is not available for my Region (eu-north). I fixed this but the DOM error still remains.
> 
> Log:
> 
> ```
> plugin:voice:7241 Uncaught (in promise) DOMException: The element has no supported sources.
> polly.eu-north-1.amazonaws.com/v1/speech:1     Failed to load resource: the server responded with a status of 400 (Bad Request)
> plugin:voice:7228 Error playing the audio stream: ValidationException: The selected engine is not supported in this region
>     at eval (plugin:voice:3980:51)
>     at step (plugin:voice:2925:19)
>     at Object.eval [as next] (plugin:voice:2872:14)
>     at fulfilled (plugin:voice:2843:24)
> callPolly @ plugin:voice:7228
> 2plugin:voice:7241 Uncaught (in promise) DOMException: The element has no supported sources.
> ```
> 
> Log after fixing the region issue:
> 
> ```
> plugin:voice:7241 Uncaught (in promise) DOMException: The element has no supported sources.
>   playAudio @ plugin:voice:7241
>   callback @ plugin:voice:7527
>   rj @ app.js:1
>   t.onChooseItem @ app.js:1
>   t.onChooseSuggestion @ app.js:1
>   t.selectSuggestion @ app.js:1
>   e.useSelectedItem @ app.js:1
>   (anonymous) @ app.js:1
>   e.handleKey @ app.js:1
>   e.onKeyEvent @ app.js:1
> ```
> 
> All in all, I have performed these steps:
> 
> 1. Created an AWS user with full Polly access, along with an access key.
> 2. Passed the key into Obsidian's settings.
> 3. Selected a voice that was available in Polly in AWS, so that I don't get the "voice unavailable in your region" error.
> 4. Restarted Obsidian
> 5. Tried the "Voice: Start reading the current document" command on a small file with only one paragraph.

@sebastianhuus 
I've recently added a bunch of languages for enhanced experience, however they have not been tested across all regions. 
Can you try using the default Voice: Stephen (American) or Joanna (American) ?

If you are still experiencing the same error, please provide more context: region, voice combination, did you get the plugin to work at all?

**Comment by sebastianhuus** [comment]
@chrisurf 

I never got the plugin to work. I gave it a shot after the update. The DOMException error is gone now, but I'm getting the same errors otherwise. 

I just tested with Stephen, but I see that it is unavailable in the Stockholm region when I use the web console. I then changed to Joanna and restarted Obsidian. Same error occurs in the Obsidian console when I call "Voice: Start reading the current document"

More context:
Region: Stockholm (https://polly.eu-north-1.amazonaws.com/v1/speech)
Voice(s): Stephen and Joanna
OS: This time I used Windows 10 Pro 22H2
Device: Desktop

```
polly.eu-north-1.amazonaws.com/v1/speech:1 Failed to load resource: the server responded with a status of 400 (Bad Request)

plugin:voice:7239 Error playing the audio stream: ValidationException: The selected engine is not supported in this region
    at eval (plugin:voice:3991:51)
    at step (plugin:voice:2936:19)
    at Object.eval [as next] (plugin:voice:2883:14)
    at fulfilled (plugin:voice:2854:24)
callPolly	@	plugin:voice:7239
```

The weird thing is how playing and generating audio in Polly through the web console works totally fine using the Stockholm region, but through Obsidian it gives the exception "The selected engine is not supported in this region" as noted above. I was thinking there could be a restriction on the region when using the Polly API, but I can't see anything that would point to that based on the Polly API reference.
